src/server/game/Spells/SpellEffects.cpp

```diff
@@ -4708,9 +4708,10 @@ void Spell::EffectKnockBack(SpellEffIndex effIndex)
     if (!unitTarget)
         return;
 
-    if (Creature* creatureTarget = unitTarget->ToCreature())
-        if (creatureTarget->isWorldBoss() || creatureTarget->IsDungeonBoss())
-            return;
+    if (m_caster->GetTypeId() == TYPEID_PLAYER || m_caster->GetOwnerGUID().IsPlayer() || m_caster->IsHunterPet())
```

Comment:
below my variant of Spell::EffectKnockBack() handler
hacky? yes, but works
maybe who find good common cases for all it

 ```c++
void Spell::EffectKnockBack(SpellEffectIndex eff_idx)
{
    if (!unitTarget)
        return;

    // Can't knockback target underwater
    if (unitTarget->IsInWater())
        return;

    // Can't knockback rooted / stunned targets
    if (unitTarget->hasUnitState(UNIT_STAT_ROOT | UNIT_STAT_STUNNED))
        return;

    // Can't knockback targets with disable move flag
    if (unitTarget->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE))
        return;

    // Can't knockback creatures with immune / boss flags, BG vehicles
    if (IsUnit(unitTarget))
    {
        Creature* creature = AsCreature(unitTarget);

        CreatureInfo const* info = creature->GetCreatureInfo();
        if ((info->MechanicImmuneMask & MECHANIC_INTERRUPT) ||
            (info->type_flags & CREATURE_TYPE_FLAG_BOSS_MOB)) // Immune Knockback
        {
            return;
        }

        if (creature->IsVehicle() && creature->GetMap()->IsBattleGround())
            return;
    }

    // Bladestorm should provide immunity to knockbacks
    if (unitTarget->HasAura(46924, EFFECT_INDEX_1))
        return;

    if (IsPlayer(m_caster))
    {
        // BF vehicles
        if (unitTarget->IsVehicle() && AsPlayer(m_caster)->InOutdoorPvPOrBattleField())
            return;

        // Typhoon
        if (m_spellInfo->IsFitToFamily(SPELLFAMILY_DRUID, uint64(0x100000000000000)))
        {
            if (m_caster->HasAura(62135, EFFECT_INDEX_0)) // Glyph of Typhoon
                return;
        }

        // Thunderstorm
        if (m_spellInfo->IsFitToFamily(SPELLFAMILY_SHAMAN, uint64(0x00200000000000)))
        {
            if (m_caster->HasAura(62132, EFFECT_INDEX_0)) // Glyph of Thunderstorm
                return;
        }

        // Blast Wave
        if (m_spellInfo->IsFitToFamily(SPELLFAMILY_MAGE, uint64(0x0004000000000)))
        {
            if (m_caster->HasAura(62126, EFFECT_INDEX_0)) // Glyph of Blast Wave
                return;
        }
    }

    // Instantly interrupt non melee spells being casted
    if (unitTarget->IsNonMeleeSpellCasted(true, false, true))
        unitTarget->InterruptNonMeleeSpells(true, 0, true);

    // Soaring - Test Flight chain
    switch (m_spellInfo->Id)
    {
        case 36812:
        case 37910:
        case 37940:
        case 37962:
        case 37968:
            unitTarget->RemoveAurasDueToSpell(36801);
            break;
    }

    unitTarget->KnockBackFrom(m_caster, float(m_spellInfo->EffectMiscValue[eff_idx]) / 10.0f, float(damage) / 10.0f);
}
```